src/devices/boiler.cpp

```diff
@@ -658,8 +673,43 @@ void Boiler::process_UBAEnergySupplied(std::shared_ptr<const Telegram> telegram)
 }
 
 // Heatpump power - type 0x48D
+//08 00 FF 00 03 8D 03 00 10 30 10 60 00 04 00 00 00 17 00 00 00 3C 38 0E 64 00 00 0C 33 C7 00 
+//XR1A050001   A05 Pump Heat circuit (1.0 ) 1 >> 1 & 0x01 ?
+//XR1A040001   A04 Pump Cold circuit (1.0 ) 1 & 0x1 ?
+
 void Boiler::process_HpPower(std::shared_ptr<const Telegram> telegram) {
     has_update(telegram->read_value(hpPower_, 11));
+    has_update(telegram->read_bitvalue(hpCompOn_, 3, 4));
+    has_update(telegram->read_value(hpBrinePumpSpd_, 5));
+    has_update(telegram->read_value(hpCompSpd_, 17));
+    has_update(telegram->read_value(hpCircSpd_, 4));
+    has_update(telegram->read_bitvalue(hpSwitchValve_, 0, 4));
+    has_update(telegram->read_value(hpActivity_, 7));
+
+    hpHeatingOn_ = 0;
+    hpCoolingOn_ = 0;
+    hpWwOn_ = 0;
+    hpPoolOn_ = 0;
+
```

Comment:
Do you really need these extra values? I think the information is completly included in `enum_hpactivity`

Reply:
Yes, I know they are a bit redundant, but I use them in a diagram (see below) from an influxd database.
If you think they take up too much space, or that they are redundant - I'll remove them.

<img width="404" alt="Skärmavbild 2021-09-03 kl  12 17 09" src="https://user-images.githubusercontent.com/29543733/131989936-2f17cedc-371a-4baa-b6ac-d044f5702f4a.png">

Reply:
Pls, let me know.

Reply:
I have no problem with the values, there's room enough. It's up to Michael to decide as the main reviewer

BTW great work navigating through and understanding the code. If you have any suggestions on how to improve the logic or readability let me know

Reply:
If the redundancy was intended, it's ok. I only want to mention it. 
We already have some redundant values like heatingactive and tapactiavated, but these are published as extra topic on change, whereas the boiler_data normally is published scheduled.